src/clr_bundle_add.c

```diff
@@ -37,163 +37,65 @@
-	fprintf(stderr, "Help Options:\n");
-	fprintf(stderr, "   -h, --help              Show help options\n");
-	fprintf(stderr, "   -u, --url=[URL]         RFC-3986 encoded url for version string and content file downloads\n");
-	fprintf(stderr, "   -c, --contenturl=[URL]  RFC-3986 encoded url for content file downloads\n");
-	fprintf(stderr, "   -v, --versionurl=[URL]  RFC-3986 encoded url for version string download\n");
-	fprintf(stderr, "   -P, --port=[port #]        Port number to connect to at the url for version string and content file downloads\n");
-	fprintf(stderr, "   -p, --path=[PATH...]    Use [PATH...] as the path to verify (eg: a chroot or btrfs subvol\n");
-	fprintf(stderr, "   -F, --format=[staging,1,2,etc.]  the format suffix for version file downloads\n");
-	fprintf(stderr, "   -n, --nosigcheck        Do not attempt to enforce certificate or signature checking\n");
-	fprintf(stderr, "   -I, --ignore-time       Ignore system/certificate time when validating signature\n");
-	fprintf(stderr, "   -S, --statedir          Specify alternate swupd state directory\n");
-	fprintf(stderr, "   -C, --certpath          Specify alternate path to swupd certificates\n");
-	fprintf(stderr, "   -t, --time              Show verbose time output for swupd operations\n");
-	fprintf(stderr, "   -N, --no-scripts        Do not run the post-update scripts and boot update tool\n");
-	fprintf(stderr, "   -b, --no-boot-update    Do not update the boot files using clr-boot-manager\n");
-	fprintf(stderr, "   -D, --max-parallel-downloads=[n] Set the maximum number of parallel downloads\n");
+	fprintf(stderr, "   swupd bundle-add [OPTION...] [bundle1 bundle2 (...)]\n\n");
+
+	//TODO: Add documentation explaining this command
+
+	global_print_help();
+
+	fprintf(stderr, "Options:\n");
 	fprintf(stderr, "   --skip-diskspace-check  Do not check free disk space before adding bundle\n");
 	fprintf(stderr, "\n");
 }
 
 static const struct option prog_opts[] = {
-	{ "help", no_argument, 0, 'h' },
-	{ "url", required_argument, 0, 'u' },
-	{ "contenturl", required_argument, 0, 'c' },
-	{ "versionurl", required_argument, 0, 'v' },
-	{ "port", required_argument, 0, 'P' },
 	{ "list", no_argument, 0, 'l' },
-	{ "path", required_argument, 0, 'p' },
-	{ "format", required_argument, 0, 'F' },
-	{ "nosigcheck", no_argument, 0, 'n' },
-	{ "ignore-time", no_argument, 0, 'I' },
-	{ "statedir", required_argument, 0, 'S' },
-	{ "certpath", required_argument, 0, 'C' },
-	{ "time", no_argument, 0, 't' },
-	{ "no-scripts", no_argument, 0, 'N' },
-	{ "no-boot-update", no_argument, 0, 'b' },
-	{ "max-parallel-downloads", required_argument, 0, 'D' },
 	{ "skip-diskspace-check", no_argument, &skip_diskspace_check, 1 },
-	{ 0, 0, 0, 0 }
+};
+
+static bool parse_opt(int opt, UNUSED_PARAM char *optarg)
+{
+	switch (opt) {
+	case 'l':
+		fprintf(stderr, "error: [-l, --list] option is deprecated, use\n"
```

Comment:
I see sometimes you are using _error_ like in this case and some times you are using _Error_ like in line 77 of clean.c, again I'd prefer if we start using these in a consistent manner, I think this kind of small inconsistencies change the perception of quality for the end user.

Reply:
I'll use the "Error" because it's more used in swupd in general. Thanks for noticing that.